### PR TITLE
Issue/#2019 notification UI

### PR DESCRIPTION
**Pull request by Sheikah45**
Redo error notifications to be more helpful and include link to tech help forum.

Also change some common user errors to not be an error notification and instead a warning

### CODE REVIEW

**Comment by Sheikah45** [comment]
New error notification

closed
Screenshot (220)

open
Screenshot (221)

File: src/main/java/com/faforever/client/notification/ImmediateNotificationController.java [hunk 1]
```diff
@@ -55,12 +52,15 @@ public ImmediateNotificationController setNotification(ImmediateNotification not
...
+      exceptionArea.setExpanded(false);
     } else {
       exceptionTextArea.setVisible(false);
+      exceptionArea.setVisible(false);
+      helpText.setVisible(false);
     }
 
     dialogLayout.setHeading(new Label(notification.getTitle()));
-    JavaFxUtil.runLater(() -> errorMessageView.getEngine().loadContent(notification.getText()));
+    notificationText.setText(notification.getText());
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/notification/ImmediateNotificationController.java:63]: Are you sure this isn't used by html notifications (e.g. from the server)

File: src/main/resources/i18n/messages.properties [hunk 2]
```diff
@@ -121,7 +123,7 @@ news.authoredFormat={0} on {1,date}
 chat.noOpenChats=No chats are open
 chat.joinAChannel=Join a channel
 chat.channelNamePrompt=e.g. #aeolus
-chat.error.noHashTag=Channel Name must start with # (Try #{0})
+chat.error.noHashTag=Channel Name must start with # (Try #{0}) aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
Comment (Brutus5000) [inline src/main/resources/i18n/messages.properties]: Derp.

File: src/main/java/com/faforever/client/chat/ChatController.java [hunk 3]
```diff
@@ -237,7 +237,7 @@ public void onJoinChannelButtonClicked() {
     channelNameTextField.clear();
     if (!channelName.startsWith("#")) {
       log.info("Channel name {} does not start with #", channelName);
-      notificationService.addImmediateErrorNotification(new IllegalArgumentException(), "chat.error.noHashTag", channelName);
+      notificationService.addImmediateNotification("errorTitle", "chat.error.noHashTag", channelName);
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChatController.java]: Please use a constant for "errorTitle"

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

**Comment by Brutus5000** [comment]
@Sheikah45 Also I request a "copy to clipboard" button for the stacktrace / message in the box.

File: src/main/java/com/faforever/client/notification/ImmediateNotificationController.java (hunk 1 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/notification/ImmediateNotificationController.java:63]: Good point I didn't see any but will double check. I might break it apart.

File: src/main/resources/i18n/messages.properties (hunk 2 above)
Comment (Sheikah45) [reply src/main/resources/i18n/messages.properties]: agreed

File: src/main/java/com/faforever/client/chat/ChatController.java (hunk 3 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/chat/ChatController.java]: Will probably make an addImmediateWarnNotification function instead

**Comment by Sheikah45** [comment]
New UI
Screenshot (222)

Screenshot (223)

**Comment by Sheikah45** [comment]
Hoping to merge soon

**Pull request merged**